Source/Scene/ModelExperimental/PickingPipelineStage.js

```diff
@@ -66,19 +71,49 @@ PickingPipelineStage.process = function (
   }
 };
 
+function processPickTexture(renderResources, featureTable) {
+  var shaderBuilder = renderResources.shaderBuilder;
+  shaderBuilder.addUniform(
+    "sampler2D",
+    "model_pickTexture",
+    ShaderDestination.FRAGMENT
+  );
+
+  var pickingUniforms = {
+    model_pickTexture: function () {
+      return featureTable.batchTexture.pickTexture;
+    },
+  };
+
+  renderResources.uniformMap = combine(
+    pickingUniforms,
+    renderResources.uniformMap
+  );
+
+  renderResources.pickId =
+    "((featureId < model_featuresLength) ? texture2D(model_pickTexture, featureSt) : vec4(0.0))";
+}
+
 function processInstancedPickIds(renderResources, context) {
   var instanceCount = renderResources.instanceCount;
   var pickIds = new Array(instanceCount);
   var pickIdsTypedArray = new Uint8Array(instanceCount * 4);
 
-  var modelResources = renderResources.model._resources;
+  var model = renderResources.model;
+  var featureTable = model.featureTable;
+  var modelResources = model._resources;
   for (var i = 0; i < instanceCount; i++) {
-    var pickObject = {
-      model: renderResources.model,
-      node: renderResources.runtimeNode,
-      primitive: renderResources.runtimePrimitive,
-      instanceId: i,
-    };
+    var pickObject;
+    if (defined(featureTable)) {
+      pickObject = featureTable.getFeature(i);
```

Comment:
I don't think it's correct to use `i` here. It should be based on the feature id attribute, which either comes from the accessor or constant/divisor.

Reply:
Speaking of constant/divisor, could you add support for that for feature id vertex attributes too?